crates/holochain/src/core/workflow/publish_dht_ops_workflow/publish_query.rs

```diff
@@ -108,7 +108,7 @@ where
 }
 
 /// Get the number of ops that might need to publish again in the future.
-pub fn num_still_needing_publish(txn: &Transaction) -> WorkflowResult<usize> {
+pub fn num_still_needing_publish(txn: &Transaction, agent: AgentPubKey) -> WorkflowResult<usize> {
```

Comment:
This query was thoroughly problematic 🆘 

- It was not filtering on author even though the workflow that calls it is cell scoped. That meant that an issue publishing for one cell would prevent all publish workflow instances from pausing.
- `withhold_publish` was not being checked but is checked by `get_ops_to_publish` so as soon as countersigning is used the loop will never pause again. (and because of the missing author check, that would have been all publish workflows on the conductor)
- We were never checking `receipts_complete` which is the point of this check in the first place 😃 

There are new tests covering this query. It's checked in this file to make sure it behaves sensibly in tandem with `get_ops_to_publish`. It's also checked in `unit_tests.rs` where we verify that the workflow won't run if ops are ready for publishing by another author or if we've received enough validation receipts.